src/js/profile/subcategory.js

```diff
@@ -16,120 +14,123 @@ const descriptionClass = '.sub-category-header__description';
+    this._indicators = [];
+    this._formattingConfig = formattingConfig;
+    this._indicatorHasNoData = {}
+    this.addKeyMetrics($(this.scHeaderClone), detail);
+    this.addSubCategoryHeaders(categorySectionWrapper, subcategory, detail, isFirst);
+    this.addIndicators(categorySectionWrapper, detail, subcategory);
+  }
+
+  get indicators() {
+    return this._indicators;
+  }
+
+  get formattingConfig() {
+    return this._formattingConfig;
+  }
+
+  addSubCategoryHeaders = (wrapper, subcategory, detail, isFirst) => {
+    let scHeader = this.scHeaderClone.cloneNode(true);
+
+    if (isFirst) {
+      $(wrapper).find(subcategoryHeaderClass).remove();
+    } else {
+      $(scHeader).removeClass('first');
+      $(scHeader).addClass('page-break-before');
     }
 
-    get indicators() {
-        return this._indicators;
-    }
+    $(subcategoryTitleClass, scHeader).text(subcategory);
+    $(descriptionTextClass, scHeader).text(detail.description);
 
-    get formattingConfig() {
-        return this._formattingConfig;
+    if (detail.description === '') {
+      $(descriptionClass, scHeader).addClass('hidden');
     }
 
-    addSubCategoryHeaders = (wrapper, subcategory, detail, isFirst) => {
-        let scHeader = scHeaderClone.cloneNode(true);
-
-        if (isFirst) {
-            $(wrapper).find(subcategoryHeaderClass).remove();
-        } else {
-            $(scHeader).removeClass('first');
-            $(scHeader).addClass('page-break-before');
-        }
+    wrapper.append(scHeader);
+  }
 
-        $(subcategoryTitleClass, scHeader).text(subcategory);
-        $(descriptionTextClass, scHeader).text(detail.description);
+  addIndicatorBlock(container, title, indicator, isLast) {
+    let block = new Indicator(this, container, title, indicator, isLast);
+    this.bubbleEvents(block, [
+      'profile.chart.saveAsPng', 'profile.chart.valueTypeChanged',
+      'profile.chart.download_csv', 'profile.chart.download_excel', 'profile.chart.download_json', 'profile.chart.download_kml',
+      'point_tray.subindicator_filter.filter'
+    ]);
 
-        if (detail.description === '') {
-            $(descriptionClass, scHeader).addClass('hidden');
-        }
+    return block;
+  }
 
-        wrapper.append(scHeader);
-    }
+  addHTMLBlock(container, title, html, isLast) {
+    let block = new HTMLBlock(this, container, title, html, isLast)
 
-    addIndicatorBlock(container, title, indicator, isLast) {
-        let block = new Indicator(this, container, title, indicator, isLast);
-        this.bubbleEvents(block, [
-            'profile.chart.saveAsPng', 'profile.chart.valueTypeChanged',
-            'profile.chart.download_csv', 'profile.chart.download_excel', 'profile.chart.download_json', 'profile.chart.download_kml',
-            'point_tray.subindicator_filter.filter'
-        ]);
+    return block;
+  }
 
-        return block;
+  displayNoDataMsg(wrapper, msg, sc) {
+    if (!this._indicatorHasNoData[sc]) {
```

Comment:
So this method will only display a 'no data' response if this._indicatorHasNoData is correctly set? At the moment I don't see this set anywhere